# Verifying the constraints for all seeds

Writing a simple tests to ensure that all of the outputs
from the function are correct immediately catches the error
that I introduced.

Doing so also demonstrates that the exception report, as
implemented, really isn't adequate

```python
Traceback (most recent call last):
  File "/Users/Danil/source/3rdparty/DanilSuits/chessKata-python/test/test_chessboard.py", line 21, in test_functional_core_all_seeds
    self.sut_satisifies_specification_for_all_seeds(system_under_test)
  File "/Users/Danil/source/3rdparty/DanilSuits/chessKata-python/test/test_chessboard.py", line 26, in sut_satisifies_specification_for_all_seeds
    self.pieces_satisfy_specification(pieces)
  File "/Users/Danil/source/3rdparty/DanilSuits/chessKata-python/test/test_chessboard.py", line 35, in pieces_satisfy_specification
    self.has_bishops_on_opposite_colors(pieces)
  File "/Users/Danil/source/3rdparty/DanilSuits/chessKata-python/test/test_chessboard.py", line 60, in has_bishops_on_opposite_colors
    self.check_opposite_colors(first_position, second_position)
  File "/Users/Danil/source/3rdparty/DanilSuits/chessKata-python/test/test_chessboard.py", line 64, in check_opposite_colors
    self.assertEqual(distance % 2, 1)
AssertionError: 0 != 1
```

We really should be able to see the arrangement of pieces that isn't
satisfying the check.

```python
Traceback (most recent call last):
  File "/Users/Danil/source/3rdparty/DanilSuits/chessKata-python/test/test_chessboard.py", line 21, in test_functional_core_all_seeds
    self.sut_satisifies_specification_for_all_seeds(system_under_test)
  File "/Users/Danil/source/3rdparty/DanilSuits/chessKata-python/test/test_chessboard.py", line 26, in sut_satisifies_specification_for_all_seeds
    self.pieces_satisfy_specification(pieces)
  File "/Users/Danil/source/3rdparty/DanilSuits/chessKata-python/test/test_chessboard.py", line 35, in pieces_satisfy_specification
    self.has_bishops_on_opposite_colors(pieces)
  File "/Users/Danil/source/3rdparty/DanilSuits/chessKata-python/test/test_chessboard.py", line 60, in has_bishops_on_opposite_colors
    self.check_opposite_colors(first_position, second_position, pieces)
  File "/Users/Danil/source/3rdparty/DanilSuits/chessKata-python/test/test_chessboard.py", line 64, in check_opposite_colors
    self.assertEqual(distance % 2, 1, "Bishops are on the same color: " + pieces)
AssertionError: 0 != 1 : Bishops are on the same color: BRBKRQNN
```

